src/components/Text/index.tsx

```diff
@@ -136,9 +139,11 @@ const Text = ({
     }
   }, [oneLine])
 
+  const StyledTextComponent = StyledText(as)
```

Comment:
Wuut ? That is really ugly, and kinda dangerous this

No way to have it work with the native styled as ?

Reply:
@chambo-e The problem with native one is that component doesn't accept any others props that the ones of the default tag (here it as a `div`) so `htmlFor` won't be accepted as it doesn't exist in div properties. That's the only workaround I've found for this :/